### DISCUSSION TITLE
2 GPIOs will trigger the Power ON signal

### DISCUSSION THREAD

**Discussion by Nexxudus** [arendst/Tasmota#14594] [Q&A] [answered]
Hi all, 
i'm having a problem with a ESP8266 using the current Tasmota KNX Firmware. Because i'm pretty new on using that, maybe this question is stupid and everybody know that beside me, then I would like to apologize in advance.

My Setup: 
D1 Mini with ESP8266
GPIO 2 is a WS2812 LED Strip (Configured as WS2812 in Configuration)
GPIO 4 is a AM312  Motion Sensor, Configured as Switch 1

To not use 2 Boards on the same place, i used 1 ESP8266 and connected both to this. 
In principle this also works, i can control my led strip via http and get the motion signal as POWER ON into my KNX System. 
But that is the problem. I don't want the sensor to switch the led strip on and off directly, so both should work completely independently.
The sensor should only give me a signal on the KNX bus (so output is also configured there and works).
But since the power ON/Off also switches the LED strip on or off, the led strip is also switched off after a few seconds, which should not be. 

Can you configure something via an additional rule in the console or is that not possible?

**Comment by barbudor** [comment, marked as answer]
Reading [Buttons-and-Switches](https://tasmota.github.io/docs/Buttons-and-Switches/#detach-switches-with) you will find different ways of decoupling the Switch from the power control
- [Rules](https://tasmota.github.io/docs/Buttons-and-Switches/#rules) but you need to capture each possible state. Depends on the SwitchMode you have selected. If the state is captured in a rule, it will not control power anymore.
- Decoupling using [`SwitchMode 15`](https://tasmota.github.io/docs/Buttons-and-Switches/#switchmode-15). It is a per Switch setting (different Switches on the same Tasmota can have different SwitchMode)
- Decoupling with [`SetOption114 1`](https://tasmota.github.io/docs/Buttons-and-Switches/#setoption114). It is a global setting. All Switches will be decoupled.

It may be ocnfusing why there are some many different ways to achieve similar modes so I suggest you read those section, may be experiment to see the differences in order to choose which one suits you.

**Reply by Nexxudus** [reply]
Thanks a lot, I was not aware of this SwitchModes. This is exactly what i was looking for. Many thanks for this fast help
